src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -240,13 +240,17 @@ module.exports = function<T, P, I, TI, C, CX>(
     }
 
     // Rerender
-    const instance = workInProgress.stateNode;
-    ReactCurrentOwner.current = workInProgress;
-    const nextChildren = instance.render();
-    reconcileChildren(current, workInProgress, nextChildren);
-    // Put context on the stack because we will work on children
-    if (isContextProvider(workInProgress)) {
-      pushContextProvider(workInProgress, true);
+    try {
+      const instance = workInProgress.stateNode;
+      ReactCurrentOwner.current = workInProgress;
+      const nextChildren = instance.render();
+      reconcileChildren(current, workInProgress, nextChildren);
+    } finally {
```

Comment:
try blocks prevent the containing function from being optimized. Can you find a way to solve this problem without adding a new try block? Maybe by fixing it in `unwindContext`?